homeassistant/components/ferienapidotde/binary_sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Binary sensor to indicate if today is a german vacation day or not."""
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['ferien-api==0.3.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ALL_STATE_CODES = [
+    "BW", "BY", "BE", "BB", "HB", "HH", "HE", "MV", "NI", "NW", "RP", "SL",
+    "SN", "ST", "SH", "TH"
+]
+
+ATTR_START = 'start'
+ATTR_END = 'end'
+ATTR_NEXT_START = 'next_start'
+ATTR_NEXT_END = 'next_end'
+ATTR_VACATION_NAME = 'vacation_name'
+
+CONF_STATE_CODE = 'state_code'
+
+DEFAULT_NAME = 'Vacation Sensor'
+
+# Don't rush the api. Every 12h should suffice.
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=12)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATE_CODE): vol.In(ALL_STATE_CODES),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Setups the ferienapidotde platform."""
+    state_code = config.get(CONF_STATE_CODE)
+    name = config.get(CONF_NAME)
+
+    import aiohttp.client_exceptions as exc
+    try:
+        data_object = VacationData(state_code)
+        await data_object.async_update()
+    except exc.ClientError:
+        import traceback
+        _LOGGER.warning(traceback.format_exc())
+        raise PlatformNotReady()
+
+    async_add_entities([VacationSensor(name, data_object)], True)
+
+
+class VacationSensor(BinarySensorDevice):
+    """Implementation of the vacation sensor."""
+
+    def __init__(self, name, data_object):
+        """Initialize the vacation sensor."""
+        self._name = name
+        self.data_object = data_object
+        self._state = None
+        self._state_attrs = {}
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of this device."""
+        return self._state_attrs
+
+    async def async_update(self):
+        """Update the state and state attributes."""
+        import ferien
+        await self.data_object.async_update()
+        vacations = self.data_object.data
+        current = ferien.current_vacation(vacs=vacations)
+        if current:
+            self._state = True
+            aligned_end = current.end - timedelta(seconds=1)
+            self._state_attrs = {
+                ATTR_START: current.start.strftime('%Y-%m-%d'),
```

Comment:
Timestamps should be absolute utc isoformat

Reply:
@MartinHjelmare I am not entirely sure what you mean by that... Do you mean to apply something like that: `current.start.utcnow().isoformat()`?

Reply:
If `current.start` is already a datetime object, `current.start.isoformat()` will return an isoformatted string. Make sure the time is utc.

We have this util to convert a datetime to utc:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/util/dt.py#L63

https://docs.python.org/3/library/datetime.html